### PR TITLE
[DOC] Auto-generate openapi.json

### PR DESCRIPTION
**Pull request by pavithraes**
Ref: #766

## Description

This pull request:

- Adds a script and action to auto-update the `openapi.json` static file. This file is used to generate the REST API docs.
- Adds `conda-store.sqlite` to `.gitignore`

## Pull request checklist

- [x] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

## Additional information

* While this works, there is likely a cleaner/minimal way to do this. Please let me know if you ideas to simplify this!
* Please make sure to squash-merge this PR.

## How to test

I've tested it on my fork: https://github.com/pavithraes/conda-store/pull/4

### CODE REVIEW

File: .github/workflows/generate_api_docs.yaml
Comment (pavithraes) [inline .github/workflows/generate_api_docs.yaml:11]: Remove these before merging

File: .github/workflows/generate_api_docs.yaml
Comment (peytondmurray) [inline .github/workflows/generate_api_docs.yaml:4]: Active support for 3.10 ended a year ago, any reason not to use the latest python version?

File: docusaurus-docs/scripts/generate_openapi_json.py
Comment (peytondmurray) [inline docusaurus-docs/scripts/generate_openapi_json.py:28]: I'm fine with this, or if you want to drop `main()` entirely and just call `gen_openapi_json` that seems fine too.

File: .github/workflows/generate_api_docs.yaml
Comment (peytondmurray) [reply .github/workflows/generate_api_docs.yaml:11]: Please remember to remove these.

**Review by peytondmurray — COMMENTED** [review]
Quick question: this will make a PR against the branch that someone is trying to merge, right? So the workflow looks like this:

1. I make a PR
2. This action runs, updating `openapi.json`. A PR against my branch is opened.
3. I merge the PR to my branch
4. Then I merge my branch to `main`

Is that the workflow?

**Comment by pavithraes** [comment]
> Quick question: this will make a PR against the branch that someone is trying to merge, right?

@peytondmurray No, this workflow opens a PR against `main` when a conda-store release is made. Since the docs are deployed separately, this helps us keep the API ref up-to-date with each released version.

**Comment by peytondmurray** [comment]
> @peytondmurray No, this workflow opens a PR against main when a conda-store release is made. Since the docs are deployed separately, this helps us keep the API ref up-to-date with each released version.

So my one problem with this is that doc versions that track the `main` branch will have API docs that are out of date. Are we publishing docs for the `main` branch? Do we even need to worry about this?

**Comment by pavithraes** [comment]
That's fair. The docs are currently only built against `main`.

I didn't want to generate the file on every merge to main because not all PRs affect the API.

I think we can re-generate this on updates to certain files. I'm not sure which specific files to track here, but I can update the PR to regenerate `openapi.json` on updates to anything within `conda_store_server/` for now.

**Comment by peytondmurray** [comment]
Failing unit tests look like an unspecified conda error:

```
ERROR tests/test_actions.py::test_api_get_build_installer[conda_prefix0] - conda_lock._vendor.poetry.utils._compat.CalledProcessError: Command '['mamba', 'env', 'create', '-p', '/tmp/pytest-of-runner/pytest-0/test_api_get_build_installer_c0/test-prefix', '-f', '/tmp/tmpf5eqekfj/environment.yaml']' returned non-zero exit status 1.
ERROR tests/test_actions.py::test_api_get_build_installer[conda_prefix1] - conda_lock._vendor.poetry.utils._compat.CalledProcessError: Command '['mamba', 'env', 'create', '-p', '/tmp/pytest-of-runner/pytest-0/test_api_get_build_installer_c1/test-prefix', '-f', '/tmp/tmpc1ja3ttd/environment.yaml']' returned non-zero exit status 1.
```

I'll run these again to see if they're just some flaky network issue or something real. In any case, I'll add these to the meta testing issue so we can mock them out, because we're currently calling out to conda/mamba here.

---

Separately, after a discussion with @pavithraes this PR will be modified so that autogenerated API docs are committed directly to the PR branch (as pre-commit.ci changes are) to avoid generating separate PRs to merge. We'll merge this PR once that small change is made.

**Comment by pavithraes** [comment]
@peytondmurray - this PR will now add a new commit to PRs that touch the conda-store-server files. Ready for another review :)

**Review by peytondmurray — APPROVED** [review]
Amazing, thank you!

**Pull request merged**
